### PR TITLE
Allow computing pose/twist from position/velocity

### PR DESCRIPTION
**Pull request by efernandez**
This introduces the following changes:
1. Always read wheel joint position and velocity, simplifying the previous logic.
2. If wheel position/velocity is used to either compute the pose or twist, check for NaNs (this moves the previous check for NaNs).
3. Add the `pose_from_joint_position` (default: True) and `twist_from_joint_position` (default: False) params, which replace the `position_feedback` one.
4. Rename `v_` variables into `dp_` when their meaning is "difference of position" (or incremental position) and not velocity.

### CODE REVIEW

**Comment by efernandez** [comment]
Now that @p6chen 's #17 is merged, do you have any comments on this PR?

@servos @paulbovbel @afakihcpr

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -469,51 +484,44 @@ namespace diff_drive_controller
     odometry_.setMeasCovarianceParams(k_l_, k_r_);
 
     // COMPUTE AND PUBLISH ODOMETRY
-    if (open_loop_)
+    // Read wheel joint positions and velocities:
```
Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp:487]: I'm thinking of moving this (the reading) to the very top of the `update` method, so the values are readed the closest possible to the `time` the control cycle starts.

Now there's no problem reading before setting the dynamic params.

**Comment by paulbovbel** [comment]
Nice, LGTM

**Comment by afakihcpr** [comment]
LGTM

**Comment by efernandez** [comment]
Thanks.

Note that I've just push a fix to the velocity feedback test, which now uses the param `pose_from_joint_position` instead of the old `position_feedback` one.

Merging...

**Pull request merged**